packages/deployment/lib/lambda.js

```diff
@@ -35,23 +59,28 @@ class UpdatedLambda extends Lambda {
    * @param {Object} lambda - the lambda object
    * @returns {Promise} returns the updated lambda object
    */
-  zipLambda(lambda) {
+  async zipLambda(lambda) {
     let msg = `Zipping ${lambda.local}`;
     // skip if the file with the same hash is zipped
+    // and is a valid zip file
     if (fs.existsSync(lambda.local)) {
-      return Promise.resolve(lambda);
+      try {
+        await this.validateZipFile(lambda);
+        return Promise.resolve(lambda);
+      }
+      catch (e) {
+        console.log(`${lambda.local} appears to be an invalid zip file, and will be re-built`);
+      }
     }
     const fileList = [lambda.source];
-
     if (lambda.useMessageAdapter) {
       const kesFolder = path.join(this.config.kesFolder, 'build', 'adapter');
       fileList.push(kesFolder);
       msg += ' and injecting message adapter';
     }
 
     console.log(`${msg} for ${lambda.name}`);
-
-    return utils.zip(lambda.local, fileList).then(() => lambda);
+    return utils.zip(lambda.local, fileList).then(() => lambda).catch((e) => console.log(`Error zipping ${e}`));
```

Comment:
As discussed in chat, please put the `.then()` and `.catch()` statements on their own lines, and re-throw the exception in the catch block after logging it.